packages/astro/src/core/preview/util.ts

```diff
@@ -1,11 +1,19 @@
 export function getResolvedHostForHttpServer(host: string | boolean) {
 	if (host === false) {
 		// Use a secure default
-		return '127.0.0.1';
+		return 'localhost';
```

Comment:
This is the main change to align to the Vite dev server. In nodejs <17, this defaults to 172.0.0.1. In nodejs >=17, this defaults to [::1]. So this should allow the preview server to serve in ipv6 too.